src/scroll.cc

```diff
@@ -111,11 +108,11 @@ static unsigned int scroll_count_characters_to_right(struct scroll_data* sd, con
 }
 
 static void scroll_scroll_left(struct scroll_data* sd, const std::vector<char>& buf, unsigned int amount) {
-    for (int j = 0; (j < amount) && (buf[sd->start] != '\0'); ++j) {
+    for (int i = 0; (i < amount) && (buf[sd->start] != '\0') && (sd->start < buf.size()); ++i) {
         sd->start += scroll_character_length(buf[sd->start]);
     }
 
-    if(buf[sd->start] == 0 || (unsigned) sd->start > strlen(&(buf[0]))) {
+    if (buf[sd->start] == 0 || sd->start > strlen(buf.data())) {
```

Comment:
Not sure, but as far as I understand it would be safer to use `strnlen` instead of `strlen`.